lib/rubyplot/backend/magick_wrapper.rb

```diff
@@ -19,10 +19,354 @@ class MagickWrapper < Base
+
+        },
+        heptagon: ->(draw, x, y, fill_color, border_color, size) {
+
+        },
+        octagon: ->(draw, x, y, fill_color, border_color, size) {
+
+        },
+        star_4: ->(draw, x, y, fill_color, border_color, size) {
+
+        },
+        star_5: ->(draw, x, y, fill_color, border_color, size) {
+
+        },
+        star_6: ->(draw, x, y, fill_color, border_color, size) {
+
+        },
+        star_7: ->(draw, x, y, fill_color, border_color, size) {
+
+        },
+        star_8: ->(draw, x, y, fill_color, border_color, size) {
+
+        },
+        vline: ->(draw, x, y, fill_color, border_color, size) {
+          # size is length of line
+          draw.stroke Rubyplot::Color::COLOR_INDEX[fill_color]
+          draw.line(x, y - size/2, x, y + size/2)
+        },
+        hline: ->(draw, x, y, fill_color, border_color, size) {
+          # size is length of line
+          draw.stroke Rubyplot::Color::COLOR_INDEX[fill_color]
+          draw.line(x - size/2, y, x + size/2, y)
+        },
+        omark: ->(draw, x, y, fill_color, border_color, size) {
+          # Hollow Square with truncated corners with side of square as size
+          # x, y is the center of the side
+          draw.stroke Rubyplot::Color::COLOR_INDEX[fill_color]
+          draw.fill_opacity 0
+          draw.polyline(
+            x - 3*size/8, y - size/2, x + 3*size/8, y - size/2,
+            x + 3*size/8, y - size/2, x + size/2, y - 3*size/8,
+            x + size/2, y - 3*size/8, x + size/2, y + 3*size/8,
+            x + size/2, y + 3*size/8, x + 3*size/8, y + size/2,
+            x + 3*size/8, y + size/2, x - 3*size/8, y + size/2,
+            x - 3*size/8, y + size/2, x - size/2, y + 3*size/8,
+            x - size/2, y + 3*size/8, x - size/2, y - 3*size/8,
+            x - size/2, y - 3*size/8, x - 3*size/8, y - size/2
+          )
+          draw.fill_opacity 1
+        }
+      }.freeze
+
       attr_reader :draw
 
+      def initialize
+        @axes_map = {}
+      end
+
+      def draw_x_axis(minor_ticks:, origin:, major_ticks:, major_ticks_count:)
+        if @axes_map[active_axes.object_id].nil?
+          @axes_map[@active_axes.object_id]={
+            axes: @active_axes,
+            x_origin: origin
+          }
+        else
+          @axes_map[@active_axes.object_id].merge!(x_origin: origin)
+        end
+      end
+
+      def draw_y_axis(minor_ticks:, origin:, major_ticks:, major_ticks_count:)
+        if @axes_map[@active_axes.object_id].nil?
+          @axes_map[@active_axes.object_id]={
+            axes: @active_axes,
+            y_origin: origin
+          }
+        else
+          @axes_map[@active_axes.object_id].merge!(y_origin: origin)
+        end
+      end
+
       def init_output_device file_name, device: :file
+        @canvas_width, @canvas_height = scale_figure(@canvas_width, @canvas_height)
         @draw = Magick::Draw.new
+        @axes = Magick::Draw.new
```

Comment:
I thought you were going to have one `Draw` object per subplot? Or will you implement that later?

Reply:
I thought that keeping these separate would create problems while making subplots but it does not (possibly due to within_window). I will make this change in future if needed.